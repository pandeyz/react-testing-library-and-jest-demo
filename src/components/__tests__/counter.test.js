import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Counter from '../../components/counter';

afterEach(() => {
  cleanup();
});

test('Counter initial value test', () => {
  const initCount = 0;
  render(<Counter initCount={initCount} />);
  const counterValueElem = screen.getByTestId('counter-value');
  const decrementBtn = screen.getByTestId('btn-dec');
  expect(counterValueElem).toBeInTheDocument();
  expect(counterValueElem).toHaveTextContent(0);

  // Works only when count value is 0
  if( initCount === 0 ) {
    expect(decrementBtn).toBeDisabled();
  }
});

test('Counter minus button is enabled when value is incremented', () => {
  const initCount = 0;
  render(<Counter initCount={initCount} />);
  const counterValueElem = screen.getByTestId('counter-value');
  const decrementBtn = screen.getByTestId('btn-dec');
  const incrementBtn = screen.getByTestId('btn-inc');
  
  fireEvent.click(incrementBtn);

  expect(counterValueElem).toBeInTheDocument();
  expect(counterValueElem).toHaveTextContent(1);
  expect(decrementBtn).toBeEnabled();
});

test('Counter minus button is again disabled when value is decremented to 0', () => {
  const initCount = 1;
  render(<Counter initCount={initCount} />);
  const counterValueElem = screen.getByTestId('counter-value');
  const decrementBtn = screen.getByTestId('btn-dec');
  
  fireEvent.click(decrementBtn);

  expect(counterValueElem).toBeInTheDocument();
  expect(counterValueElem).toHaveTextContent(0);
  expect(decrementBtn).toBeDisabled();
});

test('Snapshot Test', () => {
  const initCount = 0;
  const tree = renderer.create(<Counter initCount={initCount} />).toJSON();
  expect(tree).toMatchSnapshot();
});