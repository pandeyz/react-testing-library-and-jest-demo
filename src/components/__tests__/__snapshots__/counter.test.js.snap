// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
<div>
  <button
    data-testid="btn-dec"
    disabled={true}
    onClick={[Function]}
  >
    -
  </button>
  <span
    data-testid="counter-value"
    style={
      Object {
        "padding": 10,
      }
    }
  >
    0
  </span>
  <button
    data-testid="btn-inc"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
